final-project/application.py

```diff
@@ -496,6 +496,8 @@ def product():
     # If no product was found, then apologize.
     if not product_info or product_info == None:
         return apology("Sorry, we don't have information for that product")
+    elif len(product_info) > 1:
```

Comment:
That's actually kinda strange. Why wouldn't you return multiple products? If you ever only need one product, you should have an ID for it.

Reply:
It's because of the lack of javascript. This function is for the "view a product" template, whereas the search function is for the "view a list of products" template.

The reason I separate the two, is because the individual product page is more detailed, whereas the list is less-so. I could probably build more eloquent logic around alll this. It's one of those things I plan to address when the pages become more dynamic and I do not have html templates all over the place.

In the mean time, do you think I should make any changes?

Reply:
Ohhh... looking back at your comment, I now understand.  In a GET request, I should have the query parameter be an ID (unique), rather than the product name (not unique). Oops I went down a whole other path to return multiple products. Ugh.